wp-admin/options-web-app-manifest-display.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Web App manifest display field for the reading settings administration panel.
+ *
+ * As part of a core merge, the code in this file would go inside wp-admin/options-reading.php
+ *
+ * @package PWA
+ */
+
+namespace PWA_WP;
+
+/**
+ * Register web app manifest display setting.
+ */
+function register_web_app_manifest_display_setting() {
+	register_setting(
+		'reading',
+		'web_app_manifest_display',
+		array(
+			'type'              => 'string',
+			'sanitize_callback' => 'sanitize_text_field',
+		)
+	);
+}
+add_action( 'init', __NAMESPACE__ . '\register_web_app_manifest_display_setting' );
+
+/**
+ * Register web app manifest display setting field.
+ */
+function add_web_app_manifest_display_setting_field() {
+	add_settings_field(
+		'web_app_manifest_display',
+		__( 'Display', 'pwa' ),
+		__NAMESPACE__ . '\render_web_app_manifest_display_setting_field',
+		'reading'
+	);
+}
+add_action( 'admin_init', __NAMESPACE__ . '\add_web_app_manifest_display_setting_field' );
+
+/**
+ * Render web app manifest display setting field.
+ */
+function render_web_app_manifest_display_setting_field() {
+	// See: <https://developer.mozilla.org/en-US/docs/Web/Manifest/display#values>.
+	$allowed_values = array( 'fullscreen', 'standalone', 'minimal-ui', 'browser' );
```

Comment:
While probably out of scope of this UI, there is also a `display_override` property: https://developer.chrome.com/articles/display-override/